src/functions/memoize.ts

```diff
@@ -0,0 +1,38 @@
+// Import Third-party Dependencies
+import merge from "lodash.merge";
+
+// Import Internal Dependencies
+import { RC } from "../rc.js";
+
+let memoizedValue: Partial<RC> | null = null;
+
+export interface memoizeOptions {
+  /** * @default true */
+  overwrite?: boolean;
+}
+
+export interface memoizedOptions {
+  /** * @default null */
+  defaultValue?: Partial<RC>;
+}
+
+export function memoize(payload: Partial<RC>, options: memoizeOptions = {}): void {
+  const { overwrite = true } = options;
+  if (memoizedValue === null || overwrite) {
+    memoizedValue = payload;
+  }
+  else {
+    memoizedValue = merge({}, memoizedValue, payload);
+  }
+}
+
+export function memoized(options: memoizedOptions): Partial<RC> | null {
+  const { defaultValue = null } = options;
```

Comment:
Is it intended to have required options params while the only option has a default value ?
It force user to pass an empty object param as we can see in the tests

```ts
it("must return the default value (null)", () => {
    const result = memoized({});

    expect(result).to.deep.equal(memoizedPayload);
});
```

I would see `memoized()` instead. Also, it feel to be more consistent with the `memoize()` as `memoizeOptions` are **not** required.
Wdyt ?

Reply:
No you right, options should be optional